### PR TITLE
Add shard scheme constructor for non-contiguous shards

### PR DESCRIPTION
**Pull request by cw9**
take in a set of shards instead of a range

@xichen2020 @robskillington

### CODE REVIEW

**Comment by cw9** [comment]
There is no test file for scheme.go? Should I create one?

File: sharding/scheme.go [hunk 1]
```diff
@@ -29,36 +29,50 @@ import (
...
-	to   uint32
-	fn   m3db.HashFn
+	shards []uint32
+	fn     m3db.HashFn
 }
 
-// NewShardScheme creates a new sharding scheme, from and to are inclusive
-func NewShardScheme(from, to uint32, fn m3db.HashFn) (m3db.ShardScheme, error) {
+// NewShardSchemeWithRange creates a new sharding scheme with a range of consecutive shards, from and to are inclusive
+func NewShardSchemeWithRange(from, to uint32, fn m3db.HashFn) (m3db.ShardScheme, error) {
```
Comment (robskillington) [inline sharding/scheme.go]: Can you call this `NewShardSchemeFromRange`?

Comment (cw9) [inline sharding/scheme.go]: you mean still call it NewShardSchemeWithRange?

Comment (robskillington) [inline sharding/scheme.go]: No I mean call it `NewShardSchemeFromRange` instead of `NewShardSchemeWithRange`

Comment (cw9) [inline sharding/scheme.go]: Sure, done

Comment (cw9) [inline sharding/scheme.go]: lol my previous comment was a total messed up copy/paste

**Comment by robskillington** [comment]
LGTM

**Pull request merged**